### PR TITLE
Reuse config wizard forms for adding a system

### PR DESCRIPTION
**Pull request by allisonking**
Progress towards https://github.com/ethyca/fides/issues/1034

### Code Changes

Most of this is refactoring `DescribeSystemForm.tsx`, `PrivacyDeclarationForm.tsx`, `ReviewSystemForm`, and `SuccessPage` such that they can be used on other pages and are less tied to the config wizard.

* [x] Refactor forms to use our `tooltip` prop. We originally didn't do this when the designs were different, but now the designs look like they just use the regular chakra tooltip, so I refactored them all.
* [x] Give the forms an `onSuccess` prop instead of making them always advance the config wizard (so that our new form can advance its own stuff instead)
* [x] Give the forms a `systemKey` prop instead of always getting the system from the config wizard slice
* [x] Add validation schemas via Yup so we can better render when fields are required (better errors)
* [x] Remove some `any`s
* [x] Pull out the accordion that shows finished privacy declarations into its own component (`PrivacyDeclarationAccordion`), to be reused between `PrivacyDeclarationForm` and `ReviewSystemForm`
* [x] Add loading states
* [x] Fix the way some of the forms look (misaligned components) 
* [x] Cypress tests

Then there's only a few changes to plop these into the new `ManualSystemFlow` component:

* [x] Render the config wizard forms based on the current step
* [x] Hold the existing system we are working on in state

### Steps to Confirm

* [ ] Visit `/system`, click "Create new system", then "Manually generate a system"
* [ ] Go through the flow and create a system with privacy declarations

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

This doesn't do everything I would have liked to do for https://github.com/ethyca/fides/issues/1034. The big missing thing is all of the additional system fields that we want to be able to fill in through this view that we didn't need in the config wizard. Then there are other quality of life things:

* Although the buttons are clickable to go back in the process, they don't render with the proper values if a system is already half-made
* The form claims that the systems aren't registered until the end, though really they get created/updated with each step. Is this what we want? 

In any case, this PR was already getting too big, so calling it here and will continue work in another PR!

https://user-images.githubusercontent.com/24641006/190244726-ed770cd8-0b3b-4ba0-be55-e12966e81930.mov

Error messages upon touching a required field but not filling it in
image

### CODE REVIEW

File: clients/ctl/admin-ui/cypress/e2e/systems.cy.ts [hunk 1]
```diff
@@ -44,62 +44,171 @@ describe("System management page", () => {
   });
 
   describe("Can create a new system", () => {
-    it("Can create a system via yaml", () => {
+    beforeEach(() => {
```
Comment (allisonking) [inline clients/ctl/admin-ui/cypress/e2e/systems.cy.ts:47]: this diff is lousy, but basically I put the yaml tests in one describe block so the manual tests could go in a separate describe block. no changes to the yaml tests. just one huge test for the manual one.

**Comment by allisonking** [comment]
One more refactoring change I just snuck in to match the designs more:

previously:
image


now:
image

File: clients/ctl/admin-ui/src/features/config-wizard/PrivacyDeclarationForm.tsx [hunk 2]
```diff
@@ -55,20 +49,35 @@ import {
...
 
-const PrivacyDeclarationForm = ({
-  handleCancelSetup,
-}: {
-  handleCancelSetup: () => void;
-}) => {
-  const systemFidesKey = useAppSelector(selectSystemFidesKey);
-  const { data: existingSystem } = useGetSystemByFidesKeyQuery(systemFidesKey);
+const ValidationSchema = Yup.object().shape({
+  name: Yup.string().required().label("Declaration name"),
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/config-wizard/PrivacyDeclarationForm.tsx:57]: I noticed I'm able to click "add another declaration" when the name is blank. This gives the "already exists" error toast, but that's not quite right.

File: clients/ctl/admin-ui/src/features/config-wizard/SuccessPage.tsx [hunk 3]
```diff
@@ -16,24 +16,20 @@ import {
...
-  const systemFidesKey = useAppSelector(selectSystemFidesKey);
-  const dispatch = useAppDispatch();
-
-  const { data: existingSystem } = useGetSystemByFidesKeyQuery(systemFidesKey);
+interface Props {
+  systemKey: System["fides_key"];
+  onContinue: () => void;
+  onAddNextSystem: () => void;
+}
+const SuccessPage = ({ systemKey, onAddNextSystem, onContinue }: Props) => {
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/config-wizard/SuccessPage.tsx:31]: I guess it's a bit unfortunate this is called "Page" even though it's not a Next page.

File: clients/ctl/admin-ui/src/features/system/ManualSystemFlow.tsx [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
+import { Button, Grid, GridItem, Stack, Text } from "@fidesui/react";
+import { useRouter } from "next/router";
 import { useState } from "react";
 
+import { System } from "~/types/api";
+
+import DescribeSystemsForm from "../config-wizard/DescribeSystemsForm";
+import PrivacyDeclarationForm from "../config-wizard/PrivacyDeclarationForm";
+import ReviewSystemForm from "../config-wizard/ReviewSystemForm";
+import SuccessPage from "../config-wizard/SuccessPage";
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/system/ManualSystemFlow.tsx:10]: I like how you've extracted the state management from these components so that they can be reused like this. These imports make me think the dependence should probably go the other way, though: the config wizard using the components from the system feature. Could be a TODO to move them.

File: clients/ctl/admin-ui/src/features/config-wizard/PrivacyDeclarationForm.tsx [hunk 5]
```diff
@@ -55,20 +49,35 @@ import {
...
+
+interface Props {
+  systemKey: System["fides_key"];
+  onCancel: () => void;
+  onSuccess: (system: System) => void;
+}
+
+const PrivacyDeclarationForm = ({ systemKey, onCancel, onSuccess }: Props) => {
+  const { data: existingSystem, isLoading: isLoadingSystem } =
+    useGetSystemByFidesKeyQuery(systemKey);
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/config-wizard/PrivacyDeclarationForm.tsx:75]: With these components having most of their state dependence extracted, I would be a bit tempted to remove these queries as well: instead of taking the key and querying the system, take the system itself and the parent does the query. Not a big deal, though, seeing as that data is cached.

File: clients/ctl/admin-ui/cypress/e2e/systems.cy.ts (hunk 1 above)
Comment (ssangervasi) [reply clients/ctl/admin-ui/cypress/e2e/systems.cy.ts:47]: "Hide whitespace changes" is our friend

Screen Shot 2022-09-15 at 11 55 44

File: clients/ctl/admin-ui/src/features/common/form/inputs.tsx [hunk 6]
```diff
@@ -131,10 +132,9 @@ export const CustomSelect = ({
         >
           <Select
             options={options}
-            onBlur={(option) => {
-              if (option) {
-                field.onBlur(props.name);
-              }
+            onBlur={(e) => {
+              setTouched({ ...touched, [field.name]: true });
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/common/form/inputs.tsx:136]: Why is this necessary? My guess is that formik's `onBlur` will handle this, and we just need to use [validateOnBlur](https://formik.org/docs/api/formik#validateonblur-boolean).

File: clients/ctl/admin-ui/src/features/system/ManualSystemFlow.tsx (hunk 4 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/system/ManualSystemFlow.tsx:10]: yeah I was starting to think that as I was working on this. felt it even more in https://github.com/ethyca/fides/pull/1082. I'll make a ticket to refactor after this PR and https://github.com/ethyca/fides/pull/1082 are in 👍

File: clients/ctl/admin-ui/src/features/config-wizard/PrivacyDeclarationForm.tsx (hunk 5 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/config-wizard/PrivacyDeclarationForm.tsx:75]: good idea!

File: clients/ctl/admin-ui/src/features/common/form/inputs.tsx (hunk 6 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/common/form/inputs.tsx:136]: `validateOnBlur` defaults to true so we are already using it, and it does work on the text input fields, but not on these slightly more bespoke `select` fields (which wrap around react-select). The select fields wouldn't render their error on blur, and when I dug into it more, it was because of [this line](https://github.com/ethyca/fides/blob/574b5a8c70f9b2b8cc46082a8c19eda057480c0a/clients/ctl/admin-ui/src/features/common/form/inputs.tsx#L117) which will only show the error if there is an error AND the field has been touched. debugging showed that the field wasn't being marked as touched, so I added this call to `setTouched` which fixed things.

not sure if there's something else going on that is preventing `setTouched` from being called automatically by formik though

File: clients/ctl/admin-ui/src/features/config-wizard/SuccessPage.tsx (hunk 3 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/config-wizard/SuccessPage.tsx]: changed the name!

File: clients/ctl/admin-ui/src/features/system/ManualSystemFlow.tsx (hunk 4 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/system/ManualSystemFlow.tsx]: 🎫  https://github.com/ethyca/fides/issues/1084

File: clients/ctl/admin-ui/src/features/config-wizard/SystemRegisterSuccess.tsx [hunk 7]
```diff
@@ -110,4 +110,4 @@ const SuccessPage = () => {
     </chakra.form>
   );
 };
-export default SuccessPage;
+export default SystemRegisterSuccess;
```
Comment (LKCSmith) [inline clients/ctl/admin-ui/src/features/config-wizard/SystemRegisterSuccess.tsx:113]: Like this name a lot more ❤️ Thank you!

**Review by LKCSmith — APPROVED** [review]
✅
